### ISSUE TITLE
Support Automerge.Text without immediately binding to doc

### ISSUE THREAD

**Issue by ahixon** [automerge/automerge#166]
It's not currently possible to create an instance of Automerge.Text first and then bind it to a document later. For example:

```javascript
Automerge.change(savedDoc, doc => {
   const text = new Automerge.Text();
   // later
   doc.someText = text;
});
```

does not work correctly, but 

```javascript
Automerge.change(savedDoc, doc => {
   doc.someText = new Automerge.Text();
});
```

works just fine.

**Comment by pvh** [comment]
I'm looking at this bug today. I think there's an issue where an extra proxy is being interposed and an internal type check is failing as a result. 

The issue (mostly for @ept) is that at line context.js#69 the value is NOT an instanceof Text, though the value.constructor is Text. I'm not quite sure how to fix that yet (or entirely sure why) but I'm working towards a solution/workaround.

**Comment by pvh** [comment]
For now, it seems you can work around this issue by changing the check on that line to `value.constructor.name === "Text"`. I'm sure we can find a better fix.

**Comment by ept** [comment]
@ahixon Could you please be more specific — what do you mean exactly with "it does not work correctly"? Are you getting an exception or seeing broken behaviour?

It is true that you cannot call e.g. `.insertAt(0, 'a')` on the `Automerge.Text` object without assigning it to an object in the document, because `Automerge.Text` is read-only by default, and it only gains its ability to be mutated by being part of a document. It might be possible to change that if the current API is too confusing.

However, I don't see what this has got to do with the `instanceof Text` check, @pvh. In my test, if you assign an `Automerge.Text` object inside a change callback, the `instanceof Text` check does return true, regardless of whether you assign the object directly, or whether you first assign it to a variable and then assign that variable to a field in the document. Could you please give me a code sample where this check is failing?

**Comment by ept** [comment]
As I understand it, this is a duplicate of #180, which is now fixed by #181. The `value.constructor.name === "Text"` thing seems to be a red herring. So I'm closing this issue; please reopen if I have misunderstood something.
